### PR TITLE
[#168] Additional fields added to the Issued AC

### LINKED ISSUE #168: AIK Certificate support for additional fields
The [Attestation (AIK) Certificate ](https://trustedcomputinggroup.org/wp-content/uploads/IWG_CMC_Profile_Cert_Enrollment_v1_r7.pdf)has several mandatory and optional fields currently not shown on the Issued Attestation Certificate page.These include:

* TPM Model
* Platform Model
* TPM Specification
* Platform Specification

and MAY include the following optional fields:

* TPM Assertions
* Platform Assertions 
* Policy Reference
* Revocation Locator 

The ACA may be limited by the fields present in the Endorsement or Platform Certificate, but should be capable of displaying the optional fields.

This may require adding the fields to the Attestation Identity Certificate being generated by the ACA if its not there already.

### PR DESCRIPTION
**Pull request by cyrus-dev**
Credential Type Label     MUST 
Public AIK  MUST 
TPM Model MUST
Platform Model MUST 
Issuer MUST
TPM Specification MUST 
Platform Specification  MUST 
Signature Value MUST 
Identity Label SHOULD 
TPM Assertions MAY 
Platform Assertions MAY 
Validity Period MAY
Policy Reference MAY 
Revocation Locator MAY 

From the TCG specification, these are the elements that should be associated with an AIC and their associated state.  Generally stuff that is labelled as MAY will not appear on the page as blank if they aren't available (the field is hidden), everything else will be listed but appear empty.

Closes #168

### CODE REVIEW

File: HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java [hunk 1]
```diff
@@ -436,15 +436,44 @@ public int compare(final PlatformCredential obj1,
 
             // add endorsement credential ID if not null
             if (certificate.getEndorsementCredential() != null) {
-                data.put("endorsementID",
-                        certificate.getEndorsementCredential().getId().toString());
+                EndorsementCredential ek = certificate.getEndorsementCredential();
+                certificate.getEndorsementCredential().getId().toString();
```
Comment (chubtub) [inline HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java]: Is this line necessary?

File: HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java [hunk 2]
```diff
@@ -799,6 +839,22 @@ public TBBSecurityAssertion getTBBSecurityAssertion()
         return null;
     }
 
+    /**
+     *
+     * @param attributeSequence The sequence associated with 2.23.133.2.23
+     */
+    private void gatTCGCredentialSpecification(final ASN1Sequence attributeSequence) {
```
Comment (busaboy1340) [inline HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java]: Just nit picking, "gatTCG..." or "getTCG..."?

File: HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java [hunk 3]
```diff
@@ -728,7 +765,10 @@ public void setComponentFailures(final String componentFailures) {
                             new PlatformConfigurationV2(attributeSequence));
                     break;
                 case TCG_PLATFORM_SPECIFICATION:
+                    // handled in parseFields
+                    break;
                 case TCG_CREDENTIAL_SPECIFICATION:
+                    gatTCGCredentialSpecification(attributeSequence);
```
Comment (busaboy1340) [inline HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java]: Just nit picking, "gatTCG..." or "getTCG..."?

File: HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java (hunk 1 above)
Comment (cyrus-dev) [reply HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java]: No, forgot to take it out.

File: HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java (hunk 2 above)
Comment (cyrus-dev) [reply HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java]: Fixed.

File: HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java (hunk 3 above)
Comment (cyrus-dev) [reply HIRS_Utils/src/main/java/hirs/data/persist/certificate/PlatformCredential.java]: Fixed.

**Comment by cyrus-dev** [comment]
The last 2 system tests are failures that were fixed on a different branch and do not reflect the changes made for this pull request.

**Pull request merged**